File: apps/frontend/src/shared/api/aiChat.ts
import { api } from './client'

export interface ChatMessage {
  role: 'user' | 'assistant' | 'system'
  content: string
}

export interface SendChatMessageRequest {
  provider_id: string
  messages: ChatMessage[]
  model?: string
  stream?: boolean
}

export interface SendChatMessageResponse {
  message: ChatMessage
  model: string
}

export const aiChatApi = {
  /**
   * Отправить сообщение в AI чат
   */
  async sendMessage(
    providerId: string,
    messages: ChatMessage[],
    model?: string
  ): Promise<SendChatMessageResponse> {
    const response = await api.post('/ai/chat', {
      provider_id: providerId,
      messages,
      model: model || 'llama3.2',
      stream: false,
    })
    return response.data.data
  },
}

File: apps/backend/internal/dto/ai_chat.go
package dto

// ChatMessage представляет одно сообщение в чате
type ChatMessage struct {
	Role    string `json:"role"` // "user" | "assistant" | "system"
	Content string `json:"content"`
}

// SendChatMessageRequest - запрос на отправку сообщения в AI чат
type SendChatMessageRequest struct {
	ProviderID string        `json:"provider_id" validate:"required"`
	Messages   []ChatMessage `json:"messages" validate:"required,min=1"`
	Model      string        `json:"model"`
	Stream     bool          `json:"stream"`
}

// SendChatMessageResponse - ответ от AI чата
type SendChatMessageResponse struct {
	Message ChatMessage `json:"message"`
	Model   string      `json:"model"`
}

// GetModelsResponse - список доступных моделей
type GetModelsResponse struct {
	Models []string `json:"models"`
}
